pennylane/qchem/molecule.py

```diff
@@ -115,13 +115,13 @@ def __init__(
             l = [i[0] for i in self.basis_data]
 
         if alpha is None:
-            alpha = [np.array(i[1], requires_grad=False) for i in self.basis_data]
+            alpha = [pnp.array(i[1], requires_grad=False) for i in self.basis_data]
```

Comment:
If they're not trainable, why do they need to be autograd arrays?

Reply:
These would be trainable when one uses the differentiable Hartree-Fock workflow.

Reply:
Yep! When `alpha` is `None`, we assume it to be non-differentiable, so we use the `requires_grad=False`. But if users provide it on their own like this, `requires_grad=False` would be true. For example, 
![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/c66737e3-80f7-43e8-b74d-774ee52f4a57)

Reply:
all I know is I tried to use vanilla numpy in this file, and tests failed

Reply:
I guess out of scope for now.  But a vanilla numpy array is not trainable and compatible with all interfaces.